nimodinst system tests fail on Python 2

On Python 2, getting a string attribute returns a unicode. On Python 3 it returns a str.
This is correct behavior from NI-ModInst, but the failing tests expect str.


```
Z:\>python -m pytest src\nimodinst\system_tests\
============================= test session starts =============================
platform win32 -- Python 2.7.13, pytest-3.2.3, py-1.4.34, pluggy-0.4.0 -- C:\Python27\python.exe
cachedir: .cache
rootdir: Z:\, inifile: tox.ini
collected 15 items

src/nimodinst/system_tests/test_system_nimodinst.py::test_bad_device_family PASSED
src/nimodinst/system_tests/test_system_nimodinst.py::test_no_device_family PASSED
src/nimodinst/system_tests/test_system_nimodinst.py::test_device_family_string_with_dashes PASSED
src/nimodinst/system_tests/test_system_nimodinst.py::test_device_family_string_without_dashes PASSED
src/nimodinst/system_tests/test_system_nimodinst.py::test_int_attribute_error_on_non_existant_device PASSED
src/nimodinst/system_tests/test_system_nimodinst.py::test_string_attribute_error_on_non_existant_device PASSED
src/nimodinst/system_tests/test_system_nimodinst.py::test_device_name_attribute FAILED
src/nimodinst/system_tests/test_system_nimodinst.py::test_device_model_attribute FAILED
src/nimodinst/system_tests/test_system_nimodinst.py::test_serial_number_attribute FAILED
src/nimodinst/system_tests/test_system_nimodinst.py::test_bus_number_attribute PASSED
src/nimodinst/system_tests/test_system_nimodinst.py::test_chassis_number_attribute PASSED
src/nimodinst/system_tests/test_system_nimodinst.py::test_max_pciexpress_link_width_attribute PASSED
src/nimodinst/system_tests/test_system_nimodinst.py::test_pciexpress_link_width_attribute PASSED
src/nimodinst/system_tests/test_system_nimodinst.py::test_slot_number_attribute PASSED
src/nimodinst/system_tests/test_system_nimodinst.py::test_socket_number_attribute PASSED

================================== FAILURES ===================================
_________________________ test_device_name_attribute __________________________

    def test_device_name_attribute():
        with nimodinst.Session('') as session:
            assert len(session) > 0, 'Must have hardware for ModInst tests to be valid.'
>           assert isinstance(session[0].device_name, str)
E           AssertionError: assert False
E            +  where False = isinstance('PXI1Slot8', str)
E            +    where 'PXI1Slot8' = <nimodinst.session.Device object at 0x000000000439B4E0>.device_name

src\nimodinst\system_tests\test_system_nimodinst.py:52: AssertionError
_________________________ test_device_model_attribute _________________________

    def test_device_model_attribute():
        with nimodinst.Session('') as session:
            assert len(session) > 0, 'Must have hardware for ModInst tests to be valid.'
            assert len(session[0].device_model) > 0
>           assert isinstance(session[0].device_model, str)
E           AssertionError: assert False
E            +  where False = isinstance('NI PXIe-4145', str)
E            +    where 'NI PXIe-4145' = <nimodinst.session.Device object at 0x00000000043B2160>.device_model

src\nimodinst\system_tests\test_system_nimodinst.py:60: AssertionError
________________________ test_serial_number_attribute _________________________

    def test_serial_number_attribute():
        with nimodinst.Session('') as session:
            assert len(session) > 0, 'Must have hardware for ModInst tests to be valid.'
            pattern = r'^[0-9A-F]+$'
>           assert isinstance(session[0].serial_number, str)
E           AssertionError: assert False
E            +  where False = isinstance('0173C4F6', str)
E            +    where '0173C4F6' = <nimodinst.session.Device object at 0x00000000043AEA58>.serial_number

src\nimodinst\system_tests\test_system_nimodinst.py:69: AssertionError
===================== 3 failed, 12 passed in 0.93 seconds =====================

```

It's possible that other system tests have similar issues. #564 would catch that.
It's also a concern that we don't catch this in our unit tests.

--- a/generated/nifake/tests/test_session.py
+++ b/generated/nifake/tests/test_session.py
@@ -2,6 +2,7 @@ import matchers
 import math
 import mock_helper
 import nifake
+import six
 import warnings
 
 from mock import patch
@@ -324,10 +325,7 @@ class TestSession(object):
             assert isinstance(result_array, list)
             assert isinstance(result_array[0], float)
             assert result_string == string_val
-            try:
-                assert isinstance(result_string, basestring)
-            except NameError:
-                assert isinstance(result_string, str)
+            assert isinstance(result_string, six.text_type)
             assert self.patched_library.niFake_ReturnMultipleTypes.call_count == 2
 
     def test_multiple_array_types(self):

--- a/src/nifake/tests/test_session.py
+++ b/src/nifake/tests/test_session.py
@@ -2,6 +2,7 @@ import matchers
 import math
 import mock_helper
 import nifake
+import six
 import warnings
 
 from mock import patch
@@ -324,10 +325,7 @@ class TestSession(object):
             assert isinstance(result_array, list)
             assert isinstance(result_array[0], float)
             assert result_string == string_val
-            try:
-                assert isinstance(result_string, basestring)
-            except NameError:
-                assert isinstance(result_string, str)
+            assert isinstance(result_string, six.text_type)
             assert self.patched_library.niFake_ReturnMultipleTypes.call_count == 2
 
     def test_multiple_array_types(self):

--- a/src/nimodinst/system_tests/test_system_nimodinst.py
+++ b/src/nimodinst/system_tests/test_system_nimodinst.py
@@ -2,6 +2,7 @@
 
 import nimodinst
 import re
+import six
 
 
 def test_bad_device_family():
@@ -49,7 +50,7 @@ def test_string_attribute_error_on_non_existant_device():
 def test_device_name_attribute():
     with nimodinst.Session('') as session:
         assert len(session) > 0, 'Must have hardware for ModInst tests to be valid.'
-        assert isinstance(session[0].device_name, str)
+        assert isinstance(session[0].device_name, six.text_type)
         assert len(session[0].device_name) > 0  # device name must be at least 1 character
 
 
@@ -57,7 +58,7 @@ def test_device_model_attribute():
     with nimodinst.Session('') as session:
         assert len(session) > 0, 'Must have hardware for ModInst tests to be valid.'
         assert len(session[0].device_model) > 0
-        assert isinstance(session[0].device_model, str)
+        assert isinstance(session[0].device_model, six.text_type)
         pattern = r'(NI )?[A-Z]+e?-\d\d\d\d'
         assert re.search(pattern, session[0].device_model) is not None  # NI Model numbers are generally "NI PXIe-2532", but might also be "USB-2532"
 
@@ -66,7 +67,7 @@ def test_serial_number_attribute():
     with nimodinst.Session('') as session:
         assert len(session) > 0, 'Must have hardware for ModInst tests to be valid.'
         pattern = r'^[0-9A-F]+$'
-        assert isinstance(session[0].serial_number, str)
+        assert isinstance(session[0].serial_number, six.text_type)
         assert (len(session[0].serial_number) == 0) | (re.search(pattern, session[0].serial_number) is not None)  # NI Serial numbers hex unless it is simulated than it is 0
 
 

--- a/tox.ini
+++ b/tox.ini
@@ -123,6 +123,7 @@ deps =
     system_tests: enum34;python_version<"3.4"
     system_tests: pytest
     system_tests: coverage
+    system_tests: six
     nidcpower_system_tests: enum34;python_version<"3.4"
     nidcpower_system_tests: pytest
     nidcpower_system_tests: nidcpower

--- a/tox.ini
+++ b/tox.ini
@@ -102,6 +102,7 @@ deps =
     test: coverage
     test: mock
     test: mako
+    test: six
     build_test: pytest
     build_test: coverage
     build_test: mako
@@ -146,6 +147,7 @@ deps =
     nimodinst_system_tests: pytest
     nimodinst_system_tests: nimodinst
     nimodinst_system_tests: pytest-json
+    nimodinst_system_tests: six
 
 
 whitelist_externals =

